src/jit/codegencommon.cpp

```diff
@@ -4888,6 +4888,82 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
 #pragma warning(pop)
 #endif
 
+
+/*****************************************************************************
+* If any Vector3 args are on stack and they are not pass-by-ref, the upper 32bits
+* must be cleared to zeroes. The native compiler doesn't clear the upper bits
+* and there is no way to know if the caller is native or not. So, the upper
+* 32 bits of Vector argument on stack are always cleared to zero.
+*/
+#ifdef FEATURE_SIMD
+void    CodeGen::genClearStackVec3ArgUpperBits()
+{
+#ifdef DEBUG
+    if (verbose)
+        printf("*************** In genClearStackVec3ArgUpperBits()\n");
+#endif
+
+#if !defined(_TARGET_XARCH_)
+    return;
+#endif // !_TARGET_XARCH_
```

Comment:
+ @jkotas 

I had a chat with @jkotas regarding this.  According JKotas, deviating from native calling convention would mean that JIT is using a custom calling convention.  Not just in PInvoke/RPInvoke code paths even VM needs to be updated.  As an example, Jkotas mentioned calling any method through reflection that takes Vector3 args would need VM to clear upper 4-bytes before calling the target method.

As per JKotas, implementing a custom calling convention for Vector3 would be a slippery path.  Traditionally, JIT always implemented platform calling convention.  This would mean while reading/writing Vector3 params and Vector3 return values, Jit has to assume that the upper 4-bytes are non-zero.